### PR TITLE
Modify cache to not read file modification date

### PR DESCRIPTION
**Pull request by praveek**
- Modify CacheService to stop using fileModificationDate attribute to store cacheExpiry. 
- Clean up logs in FileSystemNamedCollection.

This change will cause a one time cache miss for currently cached content. 

Fixed `IdentityIntegrationTests .testGetExperienceCloudIdWithinPermissibleTimeOnInstall` - https://github.com/adobe/aepsdk-core-ios/issues/954

### CODE REVIEW

File: AEPServices/Sources/storage/FileSystemNamedCollection.swift [hunk 1]
```diff
@@ -49,7 +49,7 @@ class FileSystemNamedCollection: NamedCollectionProcessing {
                 do {
                     try updatedStorageData.write(to: fileUrl, options: .atomic)
                 } catch {
-                    Log.warning(label: self.LOG_TAG, "Error when writing to file: \(error)")
+                    Log.warning(label: self.LOG_TAG, "Error \(error) when writing \(key) to collection \(collectionName)")
```
Comment (sbenedicadb) [inline AEPServices/Sources/storage/FileSystemNamedCollection.swift:52]: i would vote to put our dynamic content in single quotes so it's clear to the reader exactly where the error, key, and collection name begin and end.  should we also announce `key` (e.g. - `...when writing key '\(key)'`) like we are for the other variables and in the subsequent log?

File: AEPServices/Sources/storage/FileSystemNamedCollection.swift [hunk 2]
```diff
@@ -92,13 +92,16 @@ class FileSystemNamedCollection: NamedCollectionProcessing {
...
+        }
+
+        guard let jsonResult = try? JSONSerialization.jsonObject(with: storageData) as? [String: Any] else {
+            Log.warning(label: LOG_TAG, "Failed to read dictionary from collection \(collectionName)")
+            return nil
         }
-        Log.warning(label: LOG_TAG, "Failed to get Dictionary from data store")
-        return nil
+        
+        return jsonResult
```
Comment (sbenedicadb) [inline AEPServices/Sources/storage/FileSystemNamedCollection.swift:104]: 👍

File: AEPServices/Sources/storage/FileSystemNamedCollection.swift [hunk 3]
```diff
@@ -92,13 +92,16 @@ class FileSystemNamedCollection: NamedCollectionProcessing {
             return nil
         }
 
-        if let storageData = try? Data(contentsOf: fileUrl) {
-            if let jsonResult = try? JSONSerialization.jsonObject(with: storageData) as? [String: Any] {
-                return jsonResult
-            }
+        guard let storageData = try? Data(contentsOf: fileUrl) else {
+            return nil
```
Comment (sbenedicadb) [inline AEPServices/Sources/storage/FileSystemNamedCollection.swift:96]: should we provide a message here indicating we were unable to retrieve the file contents?

File: AEPServices/Tests/services/DiskCacheServiceTests.swift [hunk 4]
```diff
@@ -184,3 +215,16 @@ class DiskCacheServiceTests: XCTestCase {
...
 }
+
+private extension Date {
+    func equal(other: Date?) -> Bool {
+        guard let other = other else {
+            return false;
+        }
+
+        // As date stores the timestamp in milliseconds, compare double value with accuracy greater than milliseconds.
+        let accuracy = 0.00001;
```
Comment (sbenedicadb) [inline AEPServices/Tests/services/DiskCacheServiceTests.swift:226]: wouldn't this just be 0.001?  my two cents is we could make it even a little more forgiving than this so we don't have random failures with this equality check

**Review by sbenedicadb — APPROVED** [review]
couple small things - take them or leave them

File: AEPServices/Sources/storage/FileSystemNamedCollection.swift (hunk 3 above)
Comment (praveek) [reply AEPServices/Sources/storage/FileSystemNamedCollection.swift:96]: This is causing a lot of false warnings during a new install.

**Review by cdhoffmann — APPROVED** [review]

**Pull request merged**